pkg/cloudlogging/puller.go

```diff
@@ -111,3 +113,47 @@ func (p *Puller) Pull(ctx context.Context, filter string, maxCount int) ([]*logg
 	}
 	return ls, nil
 }
+
+// SteamPull pulls live log entries, it stops pulling after logNum of log entries.
+func (p *Puller) SteamPull(ctx context.Context, filter string, logNum int, result chan<- []*loggingpb.LogEntry) error {
+	logger := logger.FromContext(ctx)
+	if err := retry.Do(ctx, p.retry, func(ctx context.Context) error {
+		stream, err := p.client.TailLogEntries(ctx)
+		if err != nil {
+			return retry.RetryableError(fmt.Errorf("failed to create stream: %w", err))
+		}
+		defer func() {
+			if err := stream.CloseSend(); err != nil {
```

Comment:
Let's add a member field in the Puller struct to include this stream client so that here we don't need to deal with closing the client.

Reply:
Done.